src/components/RightStatsCard.tsx

```diff
@@ -174,12 +179,39 @@ const calculateStats = (school: string, courses: Courses[]): Stats => {
     writ,
   } = getCourseStats(courses)
 
+  // Calculate how many semesters the user plans on graduating in
+  const plan = plannedGrad.split(' ', 2)
+  const totalSemesters =
+    2 * (parseFloat(plan[1]) - enrollYear) + (plan[0] === 'Fall' ? 1 : 0)
   statsObj.total = totalCredits
-  statsObj.rem = requiredCredits - totalCredits
+
+  // Set minimum remaining credits to 0
+  if (totalCredits > requiredCredits) {
+    statsObj.rem = 0
+  } else {
+    statsObj.rem = requiredCredits - totalCredits
+  }
+
+  // Logic for deciding the average remaining credits
+  if (totalSemesters <= semesters && statsObj.rem > 0) {
+    // If user still has remaining credits within time to graduate
+    statsObj.avgRem = 'Infinity - Not Possible'
+  } else if (totalSemesters < semesters && statsObj.rem <= 0) {
```

Comment:
At this point, `statsObj.rem` should be greater than or equal to 0 right?

Reply:
Yes, that would be more concise.